src/App.tsx

```diff
@@ -1,51 +1,86 @@
 import React from 'react';
+import classNames from 'classnames';
 import './App.scss';
 
-// import goodsFromServer from './goods';
+import goodsFromServer from './goods';
 
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
+type State = {
+  selectedGood: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  clearGood = () => {
+    this.setState({ selectedGood: '' });
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+
+            {selectedGood
+              ? `${selectedGood} is selected`
+              : 'No goods selected' }
+          </h1>
+
+        </header>
+
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              key={good}
+              className={classNames('Good',
+                {
+                  'Good--active': good === selectedGood,
+                })}
+            >
+              {good}
+
+              <button
```

Comment:
it is better to move everything common to goods to separate component to make your code clearer